### PR TITLE
Schema compatibility NHibernate and SQL Persistence

### PR DESCRIPTION
**Pull request by ramonsmits**
Table naming and schema difference between NHibernate and SQL Persistence.

### CODE REVIEW

File: persistence/nhibernate/index.md [hunk 1]
```diff
@@ -94,3 +94,18 @@ partial: schema
 ## Generating scripts for deployment
 
 To create scripts for execution in production without using the [installers](/nservicebus/operations/installers.md), run an install in a lower environment and then export the SQL structure. This structure can then be migrated to production.
+
+## Schema compatibility with SQL Persistence
+
+The table schema is different.
```
Comment (kbaley) [inline persistence/nhibernate/index.md]:
```suggestion
The table schema for NHibernate has a number of differences from the SQL persistence schema.
```

File: persistence/nhibernate/index.md [hunk 2]
```diff
@@ -94,3 +94,18 @@ partial: schema
...
+
+Sagas: NHibernate stores sagasas a table structure where with SQL Persistence is storing saga state as  a JSON BLOB stored more like a key/value store. Each saga type has its own table or table structure for both NHibernate and SQL Persistence.
```
Comment (kbaley) [inline persistence/nhibernate/index.md]:
```suggestion
*Sagas*: NHibernate stores sagas as a table structure while SQL Persistence stores saga state as a JSON BLOB, similar to a key/value store. Each saga type has its own table or table structure for both NHibernate and SQL Persistence.
```

File: persistence/nhibernate/index.md [hunk 3]
```diff
@@ -94,3 +94,18 @@ partial: schema
...
+
+Subscriptions: Table schemas are different. NHibernate has a single table named `Subscription` where subscriptions are stored where with SQL Persistence each logical publisher endpoint has its own subscription table `*_SubscriptionData` (endpoint name as table prefix).
```
Comment (kbaley) [inline persistence/nhibernate/index.md]:
```suggestion
*Subscriptions*: NHibernate has a single table named `Subscription` where subscriptions are stored while with SQL Persistence, each logical publisher endpoint has its own subscription table, `<endpoint_name>_SubscriptionData`.
```

File: persistence/nhibernate/index.md [hunk 4]
```diff
@@ -94,3 +94,18 @@ partial: schema
...
+
+Timeouts: NHibernate has `TimeoutEntity` table with `Endpoint` column where SQL Persistence has `*_TimeoutData` tables with the endpoint name as the table prefix.
```
Comment (kbaley) [inline persistence/nhibernate/index.md]:
```suggestion
*Timeouts*: NHibernate has a `TimeoutEntity` table with an `Endpoint` column where SQL Persistence has a `<endpoint_name>_TimeoutData` table for each endpoint.
```

File: persistence/nhibernate/index.md [hunk 5]
```diff
@@ -94,3 +94,18 @@ partial: schema
...
+
+Outbox: NHibernate has table `OutboxRecord` and requires a separate db catalog, schema, or server for each logical endpoint and SQL Persistence has table `*_OutboxData]` which has the logical endpoint name as the prefix
```
Comment (kbaley) [inline persistence/nhibernate/index.md]:
```suggestion
*Outbox*: NHibernate has an `OutboxRecord` table and requires a separate DB catalog, schema, or server for each logical endpoint. SQL Persistence has a table, `<endpoint_name>_OutboxData`, for each logical endpoint.
```

**Comment by ramonsmits** [comment]
@kbaley It doesn't really fit either. I'm not sure where to put it. I think your reasoning make sense. Maybe add a document to SQLP with the title Migrating from NHibernate to SQLP?

Or maybe we need a document in persistence called "migrating between persisters"

**Comment by kbaley** [comment]
For now, I suggest creating a separate page under SQL Persistence called "Migrating from NHibernate" with this content in it and make sure it's properly linked in the menu and other pages. A generic "migrating between persisters" page isn't as useful as one that's specific to the persister someone is migrating from/to.

**Pull request merged**